nampower/main.cpp

```diff
@@ -131,15 +131,17 @@ bool CastSpellHook(hadesmem::PatchDetourBase *detour, void *unit, int spellId, v
     // if this is a trade skill or item enchant, do nothing further
     if (spell->Effect[0] == game::SpellEffects::SPELL_EFFECT_TRADE_SKILL ||
         spell->Effect[0] == game::SpellEffects::SPELL_EFFECT_ENCHANT_ITEM ||
-        spell->Effect[0] == game::SpellEffects::SPELL_EFFECT_ENCHANT_ITEM_TEMPORARY)
-        return ret;
+        spell->Effect[0] == game::SpellEffects::SPELL_EFFECT_ENCHANT_ITEM_TEMPORARY ||
+        spell->Effect[0] == game::SpellEffects::SPELL_EFFECT_CREATE_ITEM)
+            return ret;
 
     // haven't gotten spell result from the previous cast yet, probably due to latency.
     // simulate a cancel to clear the cast bar but only when there should be a cast time
     if (!ret && castTime)
     {
         gCancelling = true;
-
+        //JT: Suggest replacing CancelSpell with InterruptSpell (the API called when moving during casting).
+        // The address of InterruptSpell needs to be dug out. It could possibly fix the sometimes broken animations.
         auto const cancelSpell = reinterpret_cast<CancelSpellT>(Offsets::CancelSpell);
```

Comment:
Have you tried this and confirmed that it works better?

Reply:
No. I have not disassembled wow.exe. This is something I suspect will work from all the logging and testing I did trying to understand how the casting works, before I could do any meaningful fixes to nampower.